okhttp/src/main/java/okhttp3/internal/concurrent/TaskLogger.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2019 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3.internal.concurrent
+
+import java.util.logging.Level
+
+internal inline fun taskLog(
+  task: Task,
+  queue: TaskQueue,
+  messageBlock: () -> String
+) {
+  if (TaskRunner.logger.isLoggable(Level.FINE)) {
+    log(task, queue, messageBlock())
+  }
+}
+
+internal inline fun <T> logElapsed(
+  task: Task,
+  queue: TaskQueue,
+  block: () -> T
+): T {
+  var startNs = -1L
+  val loggingEnabled = TaskRunner.logger.isLoggable(Level.FINE)
+  if (loggingEnabled) {
```

Comment:
Maybe a case for inverting this check and returning earlier, save a lot of indenting and checks below.

Reply:
I did it this way to avoid duplicating `block()` once inlined.